generate-key.js

```diff
@@ -3,14 +3,33 @@
     More information: https://github.com/mpgn/discord-e2e-encryption
 */
 
-async function generateKey() {
-    return window.crypto.subtle.generateKey(
-        {
-            name: "AES-GCM",
-            length: 256,
+async function generateKey(password, salt, iterations) {
+    // Based on https://8gwifi.org/docs/window-crypto-pbkdf.jsp
+
+    // Define the different part for the key
+    let saltEncoder = new TextEncoder('utf-8')
+    let saltBuffer = saltEncoder.encode(salt)
+    let encoder = new TextEncoder('utf-8')
+    let passphraseKey = encoder.encode(password)
+
+    let key = await window.crypto.subtle.importKey(
+        'raw',
+        passphraseKey,
+        {name: 'PBKDF2'},
+        false,
+        ['deriveBits', 'deriveKey']
+    )
+
+    return window.crypto.subtle.deriveKey(
+        { "name": 'PBKDF2',
+          "salt": saltBuffer,
+          "iterations": iterations,
+          "hash": 'SHA-256'
         },
+        key,
+        { "name": 'AES-CBC', "length": 256 },
```

Comment:
not a fan of CBC cipher mode, i prefer GCM :)

Reply:
It's true ^^, I also need to rewrite it because I read that we could use the deriveKey from webcrypto API to make et pbkdf2 from a password with importKey (https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveKey#PBKDF2_2).

I will rewrite it properly and it will use GCM also ^^

Reply:
nice refactoring :)